tools/vendored.py

```diff
@@ -64,6 +64,17 @@ def rewrite_importlib_resources(pkg_files, new_root):
         file.write_text(text)
 
 
+def rewrite_more_itertools(pkg_files: Path):
+    """
+    Rewrite more_itertools to remove unused more_itertools.more
+    """
+    for more_file in pkg_files.glob("more.py*"):
+        more_file.remove()
+    for init_file in pkg_files.glob("__init__.py*"):
+        text = "".join(ln for ln in init_file.lines() if "from .more " not in ln)
+        init_file.write_text(text)
```

Comment:
How about instead simply defer the import of concurrent.futures (untested)?

```suggestion
    """
    Defer import of concurrent.futures. Workaround for #3090.
    """
    more_file = pkg_files.joinpath('more.py')
    text = more_file.read_text()
    text = re.sub(r'^.*concurrent.futures.*?\n', '', text, flags=re.MULTILINE)
    text = re.sub(
        'ThreadPoolExecutor',
        '__import__("concurrent.futures").futures.ThreadPoolExecutor',
        text,
    )
    more_file.write_text(text)
```

Reply:
This also fixes the issue, updated PR with this modification